sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -786,9 +786,6 @@ std::unique_ptr<CurlSession::CurlConnection> CurlSession::GetCurlConnection(Requ
 {
   std::string const& host = request.GetHost();
 
-  // Double-check locking. Check if there is any available connection before locking mutex
-  auto& hostPoolFirstCheck = s_connectionPoolIndex[host];
-  if (hostPoolFirstCheck.size() > 0)
   {
```

Comment:
Remove the nested scope and indentation (`{`), since it is no longer necessary with the `if` block gone.

Reply:
that `{}` is to release the lock guard.